### PR TITLE
Setup spicepod for spiceai docs

### PR DESCRIPTION
**Pull request by ewgenius**

### CODE REVIEW

File: spicepod.yaml [hunk 1]
```diff
@@ -0,0 +1,104 @@
+version: v1beta1
+kind: Spicepod
+name: spice-oss-docs
+
+datasets:
+  - from: github:github.com/spiceai/docs/files/trunk
+    name: spiceai.docs
```
Comment (lukekim) [inline spicepod.yaml]: If this is the only thing, why not just put this in the default schema?

```suggestion
    name: docs
```

File: spicepod.yaml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    description: Spice.ai OSS documentation and reference, from https://docs.spiceai.org
+    metadata:
+      reference_base_url: https://docs.spiceai.org/<docs_path>
```
Comment (lukekim) [inline spicepod.yaml]:
```suggestion
    metadata:
      instructions: Documents are stored in Markdown. Always provide citations.
      reference_base_url: https://docs.spiceai.org/<docs_path>
```

File: spicepod.yaml [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+      - column: content
+        use: openai_embeddings
+        column_pk:
+          - path
+  
+  - from: github:github.com/spiceai/samples/files/trunk
+    name: spiceai.samples
+    description: Spice.ai OSS samples
+    metadata:
+      reference_base_url: https://github.com/spiceai/samples/tree/trunk/<sample_path>
```
Comment (lukekim) [inline spicepod.yaml]:
```suggestion
    metadata:
      instructions: Documents are stored in Markdown. Always provide citations.
      reference_base_url: https://github.com/spiceai/samples/tree/trunk/<sample_path>
```

File: spicepod.yaml [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+      - column: content
+        use: openai_embeddings
+        column_pk:
+          - path
+
+  - from: github:github.com/spiceai/quickstarts/files/trunk
+    name: spiceai.quickstarts
+    description: Spice.ai OSS quickstarts
+    metadata:
+      reference_base_url: https://github.com/spiceai/quickstarts/tree/trunk/<quickstart_path>
```
Comment (lukekim) [inline spicepod.yaml]:
```suggestion
    metadata:
      instructions: Documents are stored in Markdown. Always provide citations.
      reference_base_url: https://github.com/spiceai/quickstarts/tree/trunk/<quickstart_path>
```

File: spicepod.yaml [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+          - path
+
+embeddings:
+  - from: openai
+    name: openai_embeddings
+    params:
+      openai_api_key: ${secrets:OPENAI_API_KEY}
+
+  - name: hf_minilm
+    from: huggingface:huggingface.co/sentence-transformers/all-MiniLM-L6-v2
```
Comment (lukekim) [inline spicepod.yaml]:
```suggestion
```

File: spicepod.yaml [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+    params:
+      spice_tools: auto
+      openai_api_key: ${secrets:OPENAI_API_KEY}
+      system_prompt: |
+        You are an AI assistant assisting engineers with the Spice.ai OSS Project.
+
+        Always strive to be accurate, concise, and helpful in your responses.
+
+        Prefer spiceai.docs for documentation and reference information questions.
+        When generating reference links for spiceai.docs, use https://docs.spiceai.org/<docs_path> as a template and exclude `spiceaidocs/docs` prefix from docs path and `.md` file extension. Also replace `/index.md` with `/` in the path.
```
Comment (lukekim) [inline spicepod.yaml]: Move these instructions to the per dataset instructions for their specific formats.

File: spicepod.yaml [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+        
+        Prefer spiceai.samples and spiceai.quickstarts for use cases, sample code, and configuration questions. Always include links to relevant samples or quickstarts.
+
+        Use the SQL tool when:
```
Comment (lukekim) [inline spicepod.yaml]:
```suggestion
        Use the SQL tool (sql_query) when:
```

File: spicepod.yaml [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+          1. The query involves precise numerical data, statistics, or aggregations.
+          2. The user asks for specific counts, sums, averages, or other calculations.
+          3. The query requires joining or comparing data from multiple related tables.
+
+        If the SQL tool returns a query, syntax, or planning error, call the `list_datasets` tool to get the available tables and continue to refine and retry the query until it succeeds. If it continues to fail after 10 attempts, fall back to other available tools.
```
Comment (lukekim) [inline spicepod.yaml]:
```suggestion
        If the SQL tool returns a query, syntax, or planning error, call the `list_datasets` tool to get the available tables and continue to refine and retry the query until it succeeds. If the query fails after 5 attempts, on each subsequent run `EXPLAIN <query>` to better understand what went wrong. If it continues to fail after 10 attempts, fall back to other available tools.
```

File: spicepod.yaml (hunk 6 above)
Comment (ewgenius) [reply spicepod.yaml]: Moved

**Review by Sevenannn — APPROVED** [review]

**Pull request merged**